.changeset/cyan-buses-wave.md

```diff
@@ -0,0 +1,13 @@
+---
+"@fluidframework/shared-object-base": minor
+"@fluid-experimental/tree": minor
+"@fluid-experimental/tree2": minor
+---
+
+`FluidSerializer`` more strictly validates `IFluidHandles``.
+
+`FluidSerializer` now asserts on more objects that have an "IFluidHandle" member and are not `IFluidHandle`.
+One example of this is if an object implements `IProvideFluidHandle`, but is not a `IFluidHandle`, it will now assert instead of silently trying to use the "IFluidHandle" member as an `IFluidHandle`.
+
+One example of how this could be a problem is code which was passing in an `IDataStore` instead of a handle to it would silently have worked.
```

Comment:
```suggestion
One example of how this could be a problem is code which was passing in an `IDataStore` instead of a handle to it would silently encode the `IDataStore` as a handle to it.
```